### PR TITLE
[WIP] heimdal library view tracks

### PR DESCRIPTION
**Pull request by yingjieMiao**
@stkem @eishay 

added `VIEWED_LIBRARY` user event type. Shoebox is not triggering this event type yet. 

This PR gives you a preview of where I am heading.

### CODE REVIEW

File: kifi-backend/modules/heimdal/app/com/keepit/controllers/EventTrackingController.scala [hunk 1]
```diff
@@ -15,6 +16,10 @@ import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
 import scala.util.{ Failure, Success }
 
+trait UserEventHandler {
```
Comment (stkem) [inline kifi-backend/modules/heimdal/app/com/keepit/controllers/EventTrackingController.scala:19]: Nice new abstraction!

**Comment by stkem** [comment]
Good Direction!

**Comment by yingjieMiao** [comment]
@stkem @eishay 

added `VisitorEventTypes.VIEWED_LIBRARY`. This completes the tracking on Heimdal side, which makes this PR self-contained. 

I would like to open a different PR on shoebox side, really send down these events from `LibraryAnalytics`

**Comment by stkem** [comment]
I would prefer one PR with both parts if possible. Makes it easier to see what is going on. This is self contained code wise, but has no purpose without the other part (and it's not so long as to warrant a split PR)

**Comment by yingjieMiao** [comment]
@stkem   shoebox tracking added.

thank @ahsu1230 for pointing the code path

**Comment by stkem** [comment]
:+1:

**Pull request merged**